spec/sec-class-definitions-patch.html

```diff
@@ -53,225 +47,334 @@ <h1>Static Semantics: Early Errors</h1>
+            1. Let _existing_ be that PrivateElement.
+            1. Assert: <del>_field_</del><ins>_element_</ins>.[[Kind]] and _existing_.[[Kind]] are both ~accessor~.
+            1. If <del>_field_</del><ins>_element_</ins>.[[Get]] is *undefined*, then
+              1. Let _combined_ be PrivateElement { [[Key]]: <del>_field_</del><ins>_element_</ins>.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _existing_.[[Get]], [[Set]]: <del>_field_</del><ins>_element_</ins>.[[Set]] }.
+            1. Else,
+              1. Let _combined_ be PrivateElement { [[Key]]: <del>_field_</del><ins>_element_</ins>.[[Key]], [[Kind]]: ~accessor~, [[Get]]: <del>_field_</del><ins>_element_</ins>.[[Get]], [[Set]]: _existing_.[[Set]] }.
+            1. Replace _existing_ in _container_ with _combined_.
+          1. Else,
+            1. Append <del>_field_</del><ins>_element_</ins> to _container_.
+        1. Else if <del>_field_</del><ins>_element_</ins> is a ClassFieldDefinition Record, then
           1. If IsStatic of _e_ is *false*, append <del>_field_</del><ins>_element_</ins> to _instanceFields_.
           1. Else, append <del>_field_</del><ins>_element_</ins> to <del>_staticFields_</del><ins>_staticElements_</ins>.
         1. <ins>If _element_ is a ClassStaticBlockDefinition Record, then</ins>
           1. <ins>Append _element_ to _staticElements_.</ins>
-      1. <del>Set the running execution context's LexicalEnvironment to _lex_.</del>
-      1. If _className_ is not *undefined*, then</ins>
-        1. Perform _classScopeEnvRec_.InitializeBinding(_className_, _F_).
+      1. Set the running execution context's LexicalEnvironment to _lex_.
+      1. If _classBinding_ is not *undefined*, then
+        1. Perform _classScopeEnvRec_.InitializeBinding(_classBinding_, _F_).
       1. Set _F_.[[Fields]] to _instanceFields_.
-      1. If PrivateBoundIdentifiers of |ClassBody| contains a Private Name _P_ such that _P_.[[Kind]] is either `"method"` or `"accessor"` and _P_.[[Brand]] is _proto_, then
-        1. Set _F_.[[PrivateBrand]] to _proto_.
-      1. If PrivateBoundIdentifiers of |ClassBody| contains a Private Name _P_ such that _P_.[[Kind]] is either `"method"` or `"accessor"` and _P_.[[Brand]] is _F_, then
-        1. Perform ? PrivateBrandAdd(_F_, _F_).
-      1. For each element <del>_fieldRecord_</del><ins>_elementRecord_</ins> of <del>_staticFields_</del><ins>_staticElements_</ins> in List order, do
-        1. <del>Perform ? DefineField(_F_, _fieldRecord_).</del>
+      1. For each PrivateElement _method_ of _staticPrivateMethods_, do
+        1. Perform ! PrivateMethodOrAccessorAdd(_method_, _F_).
+      1. For each element <del>_fieldRecord_</del><ins>_elementRecord_</ins> of <del>_staticFields_</del><ins>_staticElements_</ins>, do
+        1. <del>Let _result_ be DefineField(_F_, _fieldRecord_).</del>
+        1. <del>If _result_ is an abrupt completion, then</del>
```

Comment:
Seems like this clause is deleted only to be re-inserted after the inserted "If _elementRecord_ is a ClassFieldDefinition Record, ... Else, ..."; consider merging the deletion and reinsertion.

Reply:
The problem is that it becomes indented in an "If". Just indenting it and inserting the "If" may not call out that such a change occurred. This will be less of an issue when I create the PR against ecma262, since the ins/del tags won't be present.

Reply:
OK, got it. I had missed the indentation.